app/src/common/shared/com/igalia/wolvic/ui/widgets/WindowWidget.java

```diff
@@ -526,6 +526,10 @@ public boolean isAddonsVisible() {
         return mViewModel.getIsAddonsVisible().getValue().get();
     }
 
+    public boolean isInWebPage() {
+        return !(isLibraryVisible() || isDownloadsVisible() || isWebAppsVisible() || isAddonsVisible() || isCurrentUriBlank());
+    }
```

Comment:
Ah :) this is exactly what I was asking for. Can't we use it in all the places where you have been adding isXVisible?